chainer/functions/split_axis.py

```diff
@@ -33,21 +32,20 @@ def forward_gpu(self, x):
         self.cdimx = xshape[self.axis]
         self.rdim = numpy.prod(xshape[self.axis + 1:])
 
-        indices = self.indices_or_sections
-        if isinstance(indices, collections.Iterable):
-            indices = list(indices)
-            indices.append(xshape[self.axis])
+        if isinstance(self.indices_or_sections, collections.Iterable):
+            ind = list(self.indices_or_sections)
+            ind.append(xshape[self.axis])
         else:
-            if xshape[self.axis] % indices:
+            sec = self.indices_or_sections
+            if xshape[self.axis] % sec:
                 raise ValueError(
                     'array split does not result in an equal division')
-            indices = six.moves.range(
-                indices, xshape[self.axis] + indices, indices)
+            ind = numpy.arange(1, sec + 1) * (xshape[self.axis] // sec)
         ys = []
         kernel = cuda.elementwise(
             _args, 'COPY(y[i] = x[idx])', 'split_fwd', preamble=_preamble)
         bi = 0
```

Comment:
It is better that the meaning of this variable should be clear from its name (like `i_prev`, `prev_i`)